presto-main/src/test/java/com/facebook/presto/cost/TestCostCalculator.java

```diff
@@ -74,7 +120,46 @@
     private static final double OFFSET_AND_IS_NULL_OVERHEAD = 13. / AVERAGE_ROW_SIZE;
     private final CostCalculator costCalculatorUsingExchanges = new CostCalculatorUsingExchanges(() -> NUMBER_OF_NODES);
     private final CostCalculator costCalculatorWithEstimatedExchanges = new CostCalculatorWithEstimatedExchanges(costCalculatorUsingExchanges, () -> NUMBER_OF_NODES);
-    private Session session = testSessionBuilder().build();
+    private final PlanFragmenter planFragmenter = new PlanFragmenter(new QueryManagerConfig());
```

Comment:
- Nullify all the fields after test class is finished
- Use `createTestTransactionManager(catalogManager)`
- Add constructor method  that accepts `transactionManager` in `MetadataManager`, and use `createTestMetadataManager(new FeaturesConfig(), transactionManager)`
- Optionally you can create similar `createTest*` methods for `nodePartitioningManager`, that creates test `nodePartitioningManager` that isn't required to be stopped, and can be simply nullified (uses direct executors instead of thread pools, etc.)

Here is the example:

```
private CostCalculator costCalculatorUsingExchanges;
    private CostCalculator costCalculatorWithEstimatedExchanges;
    private PlanFragmenter planFragmenter;
    private Session session;
    private MetadataManager metadata;
    private TransactionManager transactionManager;
    private FinalizerService finalizerService;
    private NodeScheduler nodeScheduler;
    private NodePartitioningManager nodePartitioningManager;

    @BeforeClass
    public void setUp()
    {
        costCalculatorUsingExchanges = new CostCalculatorUsingExchanges(() -> NUMBER_OF_NODES);
        costCalculatorWithEstimatedExchanges = new CostCalculatorWithEstimatedExchanges(costCalculatorUsingExchanges, () -> NUMBER_OF_NODES);
        planFragmenter = new PlanFragmenter(new QueryManagerConfig());

        session = testSessionBuilder().setCatalog("tpch").build();

        CatalogManager catalogManager = new CatalogManager();
        catalogManager.registerCatalog(createBogusTestingCatalog("tpch"));
        transactionManager = createTestTransactionManager(catalogManager);
        metadata = createTestMetadataManager(new FeaturesConfig(), transactionManager);

        finalizerService = new FinalizerService();
        finalizerService.start();
        nodeScheduler = new NodeScheduler(
                new LegacyNetworkTopology(),
                new InMemoryNodeManager(),
                new NodeSchedulerConfig().setIncludeCoordinator(true),
                new NodeTaskMap(finalizerService));
        nodePartitioningManager = new NodePartitioningManager(nodeScheduler);
    }

    @AfterClass(alwaysRun = true)
    public void tearDown()
    {
        costCalculatorUsingExchanges = null;
        costCalculatorWithEstimatedExchanges = null;
        planFragmenter = null;
        session = null;
        transactionManager = null;
        metadata = null;
        finalizerService.destroy();
        finalizerService = null;
        nodeScheduler.stop();
        nodeScheduler = null;
        nodePartitioningManager = null;
    }
```